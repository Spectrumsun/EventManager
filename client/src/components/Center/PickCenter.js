import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Display from './pickCenterProp';
import * as actions from '../../store/actions/index';

/**
 * @class Centers
 *
 * @extends {React.Component}
 */
class Centers extends Component {
  /**
   * @description run action on component mount to reload data
   *
   * @param {any} props
   *
   * @memberof Centers
   */
  componentWillMount() {
    this.props.onInitCenters();
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof Centers
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const centers = this.props.center == undefined ? [] : this.props.center.map(center =>
      (
        <Display
          key={center.id}
          id={center.id}
          centerName={center.centerName}
          about={center.about}
          address={center.address}
          city={center.city}
          image={center.imageurl}
          availability={center.availability}
          facility={center.facility}
          selectCenter={id => this.props.selectCenter(id)}
          all={center}
        />
      ));
    return (
      <div>
        <div className="container" style={{ paddingTop: '10px' }}>
          <h1 style={{ textAlign: 'center' }}>Centers</h1>
          {centers}
        </div>
      </div>
    );
  }
}

Centers.propTypes = {
  onInitCenters: PropTypes.func.isRequired,
  center: PropTypes.array.isRequired,
  selectCenter: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  center: state.centers.center
});

const mapDispatchToProps = dispatch => ({
  onInitCenters: () => dispatch(actions.initCenters())
});


export default connect(mapStateToProps, mapDispatchToProps)(Centers);
