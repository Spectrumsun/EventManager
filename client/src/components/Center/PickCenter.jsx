import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Display from './pickCenterProp';
import * as actions from '../../store/actions/index';

/**
 * @class Centers
 *
 * @extends {React.Component}
 */
class Centers extends Component {
  /**
   * @description run action on component mount to reload data
   *
   * @param {any} props
   *
   * @memberof Centers
   */
  componentWillMount() {
    this.props.onInitCenters(3, 1);
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof Centers
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const centers = this.props.center === undefined ? [] : this.props.center.map(center =>
      (
        <Display
          key={center.id}
          id={center.id}
          centerName={center.centerName.split(' ')[0]}
          about={center.about}
          address={center.address}
          city={center.city}
          image={center.imageurl}
          facility={center.facility}
          selectCenter={(id, myCenter) =>
            this.props.selectCenter(id, myCenter)}
          all={center}
        />
      ));
    return (
      <div>
        <div className="container" style={{ paddingTop: '10px' }}>
          <h1 style={{ textAlign: 'center' }}>Centers</h1>
          <div>
            <div className="fixModel">
              {centers}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Centers.propTypes = {
  onInitCenters: PropTypes.func.isRequired,
  center: PropTypes.array.isRequired,
  selectCenter: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  center: state.centers.center
});

const mapDispatchToProps = dispatch => ({
  onInitCenters: (limit, page) =>
    dispatch(actions.initCenters(3, 1))
});


export default connect(mapStateToProps, mapDispatchToProps)(Centers);
