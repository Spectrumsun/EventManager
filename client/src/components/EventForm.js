import React, { Component } from 'react';
// import { Link } from 'react-router-dom';

class EventForm extends Component {
  onSubmit(e) {
    e.preventDefault();
    //customHistory.push('/events');
  }

  render() {
    return (
      <div className="card-body" >
        <form onSubmit={this.onSubmit}>
          <div className="form-row">
            <div className="form-group col-md-6">
              <h5><label htmlFor="inputEmail4">Event Name</label></h5>
              <input type="text" className="form-control form-control-lg" required />
            </div>
            <div className="form-group col-md-6">
              <h5><label htmlFor="inputPassword4">Date</label></h5>
              <input type="date" className="form-control form-control-lg" required />
            </div>
            <div className="form-group col-md-6">
              <h5><label htmlFor="inputPassword4">Time</label></h5>
              <input type="time" className="form-control form-control-lg" required />
            </div>
          </div>
          <div className="form-group">
            <h5><label htmlFor="exampleFormControlTextarea1">Description</label></h5>
            <textarea className="form-control" id="exampleFormControlTextarea1" rows="3" placeholder="Tell people more about the event" />
          </div>
          <button type="button" className="btn btn-dark btn-sm" data-toggle="modal" data-target="#exampleModal">
			  	    Pick center
          </button>
          <div className="modal fade" id="exampleModal" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
            <div className="modal-dialog " role="document">
              <div className="modal-content">
                <div className="modal-header centerlist">
                  <h5 className="modal-title " id="exampleModalLabel">List of centers</h5>
                  <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                    <span aria-hidden="true">&times;</span>
                  </button>
                </div>
                <div className="modal-body centerlist ">
                  <h3 className="dark color">Havilah Event Centre</h3>
                  <h5 className="center">No 33 Ignobi Street Lagos</h5>
                  <h6 className="center">Facility Available</h6>
                  <ul className="list-group">
                    <li className="list-group-item centerlist" >Open Field</li>
                    <li className="list-group-item centerlist">Sound System</li>
                    <li className="list-group-item centerlist">Free Wifi Hotspot</li>
                    <li className="list-group-item centerlist">Swimming Pool</li>
                  </ul>
                  <br />
                  <button type="button" className="btn btn-primary" data-dismiss="modal">Select</button>
                </div>
                <br />
              </div>
            </div>
          </div>
          <br />
          <br />
          <button type="submit" className="btn btn-primary btn-lg">Submit</button>
        </form>
      </div>

    );
  }
}

export default EventForm;
