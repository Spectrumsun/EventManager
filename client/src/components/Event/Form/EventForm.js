import React from 'react';
import PropTypes from 'prop-types';
import PickCenter from '../../Center/PickCenter';

const eventForm = props => (
  <div className="card-body" >
    <form onSubmit={props.onSubmit}>
      <div className="form-row">
        <div className="form-group col-md-12">
          <h5><label htmlFor="inputEmail4">Event Name</label></h5>
          <input
            type="text"
            value={props.name}
            onChange={props.onChange}
            name="name"
            className="form-control form-control-lg"
          />
        </div>

        <div className="form-group col-md-6">
          <h5><label htmlFor="inputPassword4">Date</label></h5>
          <input
            type="date"
            value={props.date}
            onChange={props.onChange}
            name="date"
            className="form-control form-control-lg"
          />
        </div>

        <div className="form-group col-md-6">
          <h5><label htmlFor="inputPassword4">Time</label></h5>
          <input
            type="time"
            value={props.time}
            onChange={props.onChange}
            name="time"
            className="form-control form-control-lg"

          />
        </div>
      </div>

      <div className="form-group">
        <h5><label htmlFor="exampleFormControlTextarea1">purpose</label></h5>
        <textarea
          className="form-control"
          value={props.purpose}
          onChange={props.onChange}
          name="purpose"
          rows="3"
          placeholder="Tell people more about the event"
        />
      </div>


      <button
        type="button"
        className="btn btn-primary"
        data-toggle="modal"
        data-target=".bd-example-modal-lg"
      >
             Select Center
      </button>

      <div
        className="modal fade bd-example-modal-lg"
        tabIndex="-1"
        role="dialog"
        aria-labelledby="myLargeModalLabel"
        aria-hidden="true"
      >

        <div className="modal-dialog modal-lg">
          <div className="modal-content">
            <PickCenter selectCenter={id => props.selectCenter(id)} />
          </div>
        </div>
      </div>
      <br />
      <br />
      <button type="submit" className="btn btn-primary btn-lg">Submit</button>
    </form>
  </div>

);

eventForm.propTypes = {
  selectCenter: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  purpose: PropTypes.string.isRequired,
};


export default eventForm;

