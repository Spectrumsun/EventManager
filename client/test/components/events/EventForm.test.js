import React from 'react';
import { shallow } from 'enzyme';
import EventForm from '../../../src/components/Event/Form/EventForm';
import PickCenter from '../../../src/components/Center/PickCenter'

const props = {
  facility: ['car', 'table', 'fan'],
  onSubmit: () => { console.log('hi'); },
  onClick: () => { console.log('hi'); },
  onChange: () => { console.log('hi'); },
  selectCenter: () => { console.log('hi'); },
  getCenter: () => { console.log('hi'); },
  formValid: true,
  name: 'tomato',
  numberOfPages: { total: 2 },
  numberOfPages1: { page: 2 },
  centerName: 'yaba',
  availability: 'yes',
  startDate: '2018-12-09',
  endDate: '2018-12-10',
  time: 'uyu',
  id: 1,
  showCenterNane: 'iiir'
};

describe('<EventForm />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<EventForm {...props} />);
  });

  it('should render the HomePage', () => {
    shallow(<EventForm {...props} />);
  });

  it('should render initial layout of HomePage', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  
  it('should have div on layout', () => {
    expect(wrapper.find('div').length).toEqual(11);
  });

  it('should have  input on layout', () => {
    expect(wrapper.find('input').length).toEqual(4);
  });


  it('should have h5 on layout', () => {
    expect(wrapper.find('h5').length).toEqual(5);
  });

  it('should have ui on layout', () => {
    expect(wrapper.find('ul').length).toEqual(1);
  });

  it('should have li on layout', () => {
    expect(wrapper.find('li').length).toEqual(2);
  });

  it('should have <EventForm /> when the page loads', () => {
    expect(wrapper.find(PickCenter)).toHaveLength(1);
  });

  it('should have two image on layout', () => {
    expect(wrapper.find('button').length).toEqual(2);
  });
});
