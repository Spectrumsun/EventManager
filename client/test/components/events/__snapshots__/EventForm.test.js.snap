// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventForm /> should render initial layout of HomePage 1`] = `
Array [
  <div
    className="card-body"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-12"
        >
          <h5>
            <label
              htmlFor="inputEmail4"
            >
              Event Name
            </label>
          </h5>
          <input
            className="form-control form-control-lg"
            name="name"
            onChange={[Function]}
            required={true}
            type="text"
            value="tomato"
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <h5>
            <label
              htmlFor="inputPassword4"
            >
              Start Date
            </label>
          </h5>
          <input
            className="form-control form-control-lg"
            name="startDate"
            onChange={[Function]}
            required={true}
            type="date"
            value="2018-12-09"
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <h5>
            <label
              htmlFor="inputPassword4"
            >
              End Date
            </label>
          </h5>
          <input
            className="form-control form-control-lg"
            name="endDate"
            onChange={[Function]}
            required={true}
            type="date"
            value="2018-12-10"
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <h5>
            <label
              htmlFor="inputPassword4"
            >
              Start Time
            </label>
          </h5>
          <input
            className="form-control form-control-lg"
            name="time"
            onChange={[Function]}
            required={true}
            type="time"
            value="uyu"
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <h5>
          <label
            htmlFor="exampleFormControlTextarea1"
          >
            purpose
          </label>
        </h5>
        <textarea
          className="form-control"
          name="purpose"
          onChange={[Function]}
          placeholder="Tell people more about the event"
          required={true}
          rows="3"
          value={undefined}
        />
      </div>
      <br />
      <div
        className="form-inline"
      >
        <button
          className="btn btn-dark btn-mm"
          data-target=".bd-example-modal-lg"
          data-toggle="modal"
          id="centerSelection"
          onClick={[Function]}
          style={
            Object {
              "marginRight": "20px",
            }
          }
          type="button"
        >
          Add Center
        </button>
        iiir
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="myLargeModalLabel"
        className="modal fade bd-example-modal-lg"
        role="dialog"
        tabIndex="-1"
      >
        <div
          className="modal-dialog modal-lg"
        >
          <div
            className="modal-content"
          >
            <Connect(Centers)
              selectCenter={[Function]}
            />
            <ul
              className="pagination nav justify-content-center"
            >
              <li
                className="page-item"
                onClick={undefined}
              >
                <a
                  className="page-link"
                >
                  Previous
                </a>
              </li>
              Object {
                "total": 2,
              }
              <li
                className="page-item"
                onClick={undefined}
              >
                <a
                  className="page-link"
                >
                  Next
                </a>
              </li>
            </ul>
            <br />
          </div>
        </div>
      </div>
      <br />
      <br />
      <button
        className="btn btn-primary btn-lg"
        disabled={true}
        id="submitEvent"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
]
`;
